pkgs/servers/irc/solanum/default.nix

```diff
@@ -0,0 +1,51 @@
+{ stdenv
+, fetchFromGitHub
+, autoreconfHook
+, pkg-config
+, bison
+, flex
+, openssl
+, sqlite
+, lksctp-tools
+}:
+
+stdenv.mkDerivation rec {
+  pname = "solanum";
+  version = "unstable-2020-12-14";
+
+  src = fetchFromGitHub {
+    owner = "solanum-ircd";
+    repo = pname;
+    rev = "551e5a146eab4948ce4a57d87a7f671f2d7cc02d";
+    sha256 = "14cd2cb04w6nwck7q49jw5zvifkzhkmblwhjfskc2nxcdb5x3l96";
+  };
+
+  postPatch = ''
+    substituteInPlace include/defaults.h --replace 'PKGLOCALSTATEDIR "' '"/var/lib/solanum'
```

Comment:
This should be set via configure flags and not via sed unless really required.

Reply:
As soon as I set `--localstatedir=/var/lib/solanum` it want's to `mkdir` that.

```
test -d /var/lib/solanum/log/solanum || mkdir -p /var/lib/solanum/log/solanum
mkdir: cannot create directory '/var': Permission denied
```

Reply:
Well then we should properly patch all the other occurences of $out as well:

```
52:#define ETC_DIR "/nix/store/jjah3pw7p6r5facbkqq6iqqhj9z2zpxj-solanum-unstable-2020-12-14/etc"
258:#define HELP_DIR "/nix/store/jjah3pw7p6r5facbkqq6iqqhj9z2zpxj-solanum-unstable-2020-12-14/help"
261:#define IRCD_PREFIX "/nix/store/jjah3pw7p6r5facbkqq6iqqhj9z2zpxj-solanum-unstable-2020-12-14"
270:#define LOG_DIR "/nix/store/jjah3pw7p6r5facbkqq6iqqhj9z2zpxj-solanum-unstable-2020-12-14/logs"
307:#define MODULE_DIR "/nix/store/jjah3pw7p6r5facbkqq6iqqhj9z2zpxj-solanum-unstable-2020-12-14/modules"
355:#define PKGLIBEXECDIR "/nix/store/jjah3pw7p6r5facbkqq6iqqhj9z2zpxj-solanum-unstable-2020-12-14/bin"
358:#define PKGLOCALSTATEDIR "/nix/store/jjah3pw7p6r5facbkqq6iqqhj9z2zpxj-solanum-unstable-2020-12-14/etc"
361:#define PKGRUNDIR "/nix/store/jjah3pw7p6r5facbkqq6iqqhj9z2zpxj-solanum-unstable-2020-12-14/etc"
```

At least LOG_DIR and  PKGRUNDIR look like addtional candidates for patching.

Reply:
Done.